### ISSUE TITLE
Badge - add screen reader text

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#8354]
**Is this a new component or an extension of an existing one?**
Badge

**Describe the feature**
Similar to https://github.com/patternfly/patternfly/issues/5230, the Badge component can use screen reader text (via `pf-screen-reader` class) to help convey context rather than only using color/icons (which are only visual).

**Are there visuals for this feature? If applicable, please include examples for each state and for varying widths**
N/A

**Any other information?**
N/A

**Comment by erkanercan** [comment]
Hi @thatblindgeye,

I'm starting to work on this issue, to be clear, you want the `pf-screen-reader` to be added to the Badge classnames right?

If I can get an assignment on this issue, I'll provide a PR for it.

**Comment by erkanercan** [comment]
It seems like `pf-screen-reader` has `position: fixed` in its styles and it broke the cypress tests for the search input. I'll try to correct that issue in the same pr, help is appreciated.

**Comment by thatblindgeye** [comment]
@erkanercan Thanks for commenting! So our `pf-screen-reader` class is typically used to add visually hidden text to a component. This can be seen in our Banner component. If you inspect an alert in our [Status banner example](https://www.patternfly.org/v4/components/banner#status), you should be able to find a span with that class, but notice how it doesn't visually render on the page.

So ideally we'd want a span to render inside the badge if the prop is passed in.

**Comment by erkanercan** [comment]
Hi again @thatblindgeye, I updated the PR according to your comment, can you please review it again?
